repos/system_upgrade/el7toel8/actors/tcpwrapperscheck/actor.py

```diff
@@ -44,9 +45,13 @@ class TcpWrappersCheck(Actor):
     def process(self):
         # Consume a single TCP Wrappers message
         tcp_wrappers_messages = self.consume(TcpWrappersFacts)
-        tcp_wrappers_facts = next(tcp_wrappers_messages)
+        tcp_wrappers_facts = next(tcp_wrappers_messages, None)
         if list(tcp_wrappers_messages):
             api.current_logger().warning('Unexpectedly received more than one TcpWrappersFacts message.')
+        if not tcp_wrappers_facts:
+            raise StopActorExecutionError(
+                'Could check tcp wrappers configuration', details={'details': 'No TcpWrappersFacts found.'}
```

Comment:
The tcp_wrappers files are provided by the setup package and they should be available on every installation of RHEL7. The tcp_wrappers scanner generates this message every time.

OpenSSH does not have to be installed on all the systems, but I think the OpenSSH config scanner is made the way it returns the object even if the file is not in place and openssh not installed. I think this message does not trigger any blocking leapp reports. 

The pam is part of the system authentication so I think it is not possible to remove it and not use it. The pam scanner always returns some messages listing the pam modules used, which always make sense to scan.

Reply:
If all these three things are mandatory to be set (PAM and OpenSSH configs, TCP wrappers)  on every RHEL 7, then the upgrade should be stopped in the Checks phase, not the FactsCollection phase. And in the checks phase should create a report message for the user together with some remediation recommendation (how to proceed to be able to continue with the upgrade).

Reply:
Indeed, if there is simple way to test it, it would make sense to verify that automatically that the messages are sent even if some of the files and packages are missing.

Reply:
Not the files, but the messages are "mandatory" -- will always be sent by the leapp scanners. You can have working system without openssh and without tcp_wrappers files (but probably not without pam configuration). But that does not have anything to do with the upgrades, neither it should stop the update. The above are just sanity checks as pointed out in previous reviews by @vinzenz .

Reply:
I see, something like internal error.